mlflow/tracking/_tracking_service/client.py

```diff
@@ -155,6 +155,58 @@ def list_experiments(self, view_type=ViewType.ACTIVE_ONLY, max_results=None, pag
             view_type=view_type, max_results=max_results, page_token=page_token
         )
 
+    def search_experiments(
+        self,
+        view_type=ViewType.ACTIVE_ONLY,
+        max_results=None,
+        filter_string=None,
+        order_by=None,
+        page_token=None,
+    ):
+        """
+        Search for experiments that match the specified search query.
+
+        :param view_type: One of enum values ``ACTIVE_ONLY``, ``DELETED_ONLY``, or ``ALL``
+                          defined in :py:class:`mlflow.entities.ViewType`.
+        :param max_results: If passed, specifies the maximum number of experiments desired. If not
+                            passed, all experiments will be returned for the File and SQL backends.
+                            For the REST backend, the server will pick a maximum number of results
+                            to return.
+        :param filter_string:
+            Filter query string (e.g., ``"name = 'my_experiment'"``), defaults to searching for all
+            experiments. The following fields, comparators, and logical operators are supported.
+
+            Fields
+              - ``name``: Experiment name.
```

Comment:
Do we need to use `attribute.name` ?
and also support experiment_id field ?

Reply:
> Do we need to use attribute.name?

We don't need to. `name` is equivalent to `attribute.name`.

Reply:
ok.
and `experiment_id` should be also a valid attribute field in filter, add it in doc

Reply:
The current implementation doesn't support filtering by experiment ID.

Reply:
ok

Reply:
We can support filtering experiments with experiment ID if necessary though.